index.js

```diff
@@ -236,6 +236,29 @@ function getStsClient(region) {
   });
 }
 
+let defaultSleep = function (ms) {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+let sleep = defaultSleep;
+
+// retryAndBackoff retries with exponential backoff the promise if the error isRetryable upto maxRetries time.
+const retryAndBackoff = async (fn, isRetryable, retries = 0, maxRetries = 12, base = 50) => {
+  try {
+    return await fn();
+  } catch (err) {
+    if (!isRetryable) {
+      throw err;
+    }
+    // It's retryable, so sleep and retry.
+    await sleep( Math.max(Math.random() * (Math.pow(2, retries) * base)), );
```

Comment:
nit - seems like `Math.max` is now unnecessary 🤔 :
```suggestion
    await sleep(Math.random() * (Math.pow(2, retries) * base));
```